### PR TITLE
Don't duplicate warning strings in GetWarnings

### PR DESCRIPTION
**Pull request by Empact**
Since strStatusBar only ever holds the desired string, we can use it in the
strGUI construction as well.

### CODE REVIEW

File: src/warnings.cpp [hunk 1]
```diff
@@ -47,7 +47,7 @@ std::string GetWarnings(const std::string& strFor)
 
     if (!CLIENT_VERSION_IS_RELEASE) {
         strStatusBar = "This is a pre-release test build - use at your own risk - do not use for mining or merchant applications";
-        strGUI = _("This is a pre-release test build - use at your own risk - do not use for mining or merchant applications");
+        strGUI = _(strStatusBar);
```
Comment (MarcoFalke) [inline src/warnings.cpp]: This doesn't work with translations, imo

Comment (scravy) [reply src/warnings.cpp]: couldn't this just be `strGUI = strStatusBar`? And `strStatusBar = _(...)`

File: src/warnings.cpp [hunk 2]
```diff
@@ -39,15 +39,19 @@ void SetfLargeWorkInvalidChainFound(bool flag)
 
 std::string GetWarnings(const std::string& strFor)
 {
+    static const char* PRE_RELEASE_WARNING = "This is a pre-release test build - use at your own risk - do not use for mining or merchant applications";
```
Comment (domob1812) [inline src/warnings.cpp]: Nit:  You could make this `constexpr`, as in:

    static constexpr const char PRE_RELEASE_WARNING[] = "...";

**Review by domob1812 — COMMENTED** [review]
utACK 036821cedbf7c6010d437b467f0c8cc4a30d4659

File: src/warnings.cpp [hunk 3]
```diff
@@ -59,13 +63,13 @@ std::string GetWarnings(const std::string& strFor)
 
     if (fLargeWorkForkFound)
     {
-        strStatusBar = "Warning: The network does not appear to fully agree! Some miners appear to be experiencing issues.";
-        strGUI += (strGUI.empty() ? "" : uiAlertSeperator) + _("Warning: The network does not appear to fully agree! Some miners appear to be experiencing issues.");
+        strStatusBar = FORK_WARNING;
```
Comment (scravy) [inline src/warnings.cpp]: shouldn't these be also translated? `_(...)`

**Comment by ken2812221** [comment]
utACK 036821cedbf7c6010d437b467f0c8cc4a30d4659

**Comment by laanwj** [comment]
Please do not do this, the gettext script will not be able to pick up text if you don't directly embed it in `_()`!

**Comment by Empact** [comment]
@laanwj is there a reason not to translate the strStatusBar string?

**Comment by MarcoFalke** [comment]
@Empact I guess you wouldn't know the language setting of the client sending the request to the rpc server?

**Comment by Empact** [comment]
Gotcha! Closing as impractical.

**Comment by laanwj** [comment]
> @laanwj is there a reason not to translate the strStatusBar string?

RPC is not a user interface but a machine interface, as client-side software might be matching on messages, they should never be translated.

**Pull request closed without merging**